sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -153,6 +171,12 @@ private void startTracing(final @NonNull Activity activity) {
     }
   }
 
+  private void setMetricsForActivity(
+      final @NotNull Activity activity, final @NotNull ITransaction transaction) {
+    // set metrics and remove the slow/frozen detection since transaction is finished.
```

Comment:
`L`: Looks like this method doesn't add much. Maybe just inline it?

> and remove the slow/frozen detection since transaction is finished.

Doesn't look like removing it is happening here. Is this expected to happen by calling `setMetrics`? If so the comment is a bit disconnected (relies on a side effect of a method that isn't really named in a way that would indicate such side effect.

Reply:
It's because when you read the metrics from `FrameMetricsAggregator`, it also removes them from the internal map.
I can inline, yes.